app/notification.go

```diff
@@ -458,7 +458,39 @@ func (a *App) SendNotifications(post *model.Post, team *model.Team, channel *mod
 		message.Add("mentions", model.ArrayToJson(mentionedUsersList))
 	}
 
-	a.Publish(message)
+	var previewedPostID string
+	if val, ok := post.GetProp(model.POST_PROPS_PREVIEWED_POST).(string); ok {
+		previewedPostID = val
+	}
+	var previewedChannel *model.Channel
+	if model.IsValidId(previewedPostID) {
+		previewedPost, err := a.GetSinglePost(previewedPostID)
+		if err != nil {
+			return nil, err
+		}
+
+		channelMembers, err := a.GetChannelMembersPage(channel.Id, 0, 10000000)
+		if err != nil {
+			return nil, err
+		}
+
+		previewedChannel, err = a.GetChannel(previewedPost.ChannelId)
+		if err != nil {
+			return nil, err
+		}
+
+		for _, cm := range *channelMembers {
+			postForIndividualUser := post.Clone()
+			if !a.HasPermissionToReadChannel(cm.UserId, previewedChannel) {
+				postForIndividualUser.Metadata.Embeds[0].Data = nil
+			}
+			message.Broadcast.UserId = cm.UserId
```

Comment:
Not for this PR, but I feel like it would be really nice if this had a `UserIds` so we'd only have to potentially publish two messages here, one for users with permission and one for those without.